src/main.rs

```diff
@@ -231,51 +249,74 @@ async fn exact_search(
         .expect("[exact_search] parsing body failed");
 
     let track: Result<Track, serde_json::Error> = serde_json::from_str(&json_data);
-    match track {
+    let track_lyrics = match track {
         Ok(track) => {
-            // Find the first track with non-empty syncedLyrics
             match &track.synced_lyrics {
-                Some(lyrics) => {
-                    save_synced_lyrics(
-                        &music_dir,
-                        artist_dir,
-                        album_dir,
-                        &song_name,
-                        lyrics.clone(),
-                        successful_count,
-                    )
-                    .await;
-                }
+                Some(lyrics) => lyrics.clone(),
                 None => {
-                    println!(
-                        "[exact_search] synced lyrics unavilable {} for song {} falling back to fuzzy_search",
-                        json_data, clean_song
-                    );
-                    fuzzy_search(
-                        music_dir,
-                        artist_dir,
-                        album_dir,
-                        file_path,
-                        successful_count,
-                    )
-                    .await;
+                    match &track.plain_lyrics {
+                        Some(lyrics) => {
+                            match allow_plain_lyrics {
+                                true => lyrics.clone(),
+                                false => {
+                                    println!(
+                                        "[exact_search] synced lyrics unavailable for song {}, plain lyrics available but not allowed",
+                                        clean_song
+                                    );
+                                    // Yes, in theory having no synced lyrics with plain lyrics
+                                    // available would mean there's no point in searching again.
+                                    // But, sometimes lrclib.net returns one song for the exact
+                                    // match that doesn't have synced lyrics when another will. So
+                                    // we fall through to the fuzzy search just in case that's what
+                                    // happened here.
+                                    String::new()
+                                }
+                            }
+                        }
+                        None => {
+                            println!(
+                                "[exact_search] no lyrics available {} for song {}",
+                                json_data, clean_song
+                            );
+                            return;
```

Comment:
```suggestion
                           String::new()
```

Why not fall back to fuzzy in this case as well?

Reply:
Yea good catch, if we fall back because no synced lyrics were found we should be consistent and fall back in case we somehow found a result with no lyrics at all.